apps/allocations/app/components/App/App.js

```diff
@@ -1,187 +1,140 @@
-import { Main, AppBar, AppView, SidePanel } from '@aragon/ui'
-import PropTypes from 'prop-types'
-import React from 'react'
-import { Accounts, Payouts } from '.'
-import { NewAccount, NewAllocation } from '../Panel'
-import { networkContextType, AppTitle, AppTitleButton } from '../../../../../shared/ui'
+import React, { useState } from 'react'
+
 import { useAragonApi } from '@aragon/api-react'
-import { IdentityProvider } from '../../../../../shared/identity'
+import { AppBar, AppView, Main, SidePanel } from '@aragon/ui'
 
-class App extends React.PureComponent {
-  static propTypes = {
-    api: PropTypes.object,
-    accounts: PropTypes.arrayOf(PropTypes.object),
-  }
+import { AppTitle, AppTitleButton } from '../../../../../shared/ui'
+import { IdentityProvider } from '../../../../../shared/identity'
+import { NewAccount, NewAllocation } from '../Panel'
+import { Accounts, Payouts } from '.'
 
-  static defaultProps = {
-    network: {},
-  }
+const ASSETS_URL = './aragon-ui'
 
-  static childContextTypes = {
-    network: networkContextType,
-  }
+const nameSorter = (a, b) => a.data.name.toUpperCase() > b.data.name.toUpperCase() ? 1 : -1
 
-  state = {
-    accounts: [],
-    panel: {
-      visible: false,
-    }
-  }
+const App = () => {
+  const [ panel, setPanel ] = useState(null)
+  const { api, appState, displayMenuButton = false } = useAragonApi()
+  const { accounts = [], balances = [], entries = [], payouts = [] } = appState
 
-  getChildContext() {
-    const { network } = this.props
-    return {
-      network: {
-        type: network.type,
-      },
-    }
+  const onCreateAccount = ({ description }) => {
+    api.newAccount(description)
+    closePanel()
   }
 
-  createAccount = (account) => {
-    account.balance = 0
-    this.props.api.newAccount(account.description)
-    this.closePanel()
-    this.setState({})
-  }
-
-  submitAllocation = allocation => {
-    const emptyIntArray = new Array(allocation.addresses.length).fill(0)
-    this.props.api.setDistribution(
-      allocation.addresses,
+  const onSubmitAllocation = ({ addresses, description, payoutId, recurring, period, balance, tokenAddress }) => {
+    const emptyIntArray = new Array(addresses.length).fill(0)
+    api.setDistribution(
+      addresses,
       emptyIntArray, //[]
       emptyIntArray, //[]
       '',
-      allocation.description,
+      description,
       emptyIntArray, // Issue with bytes32 handling
       emptyIntArray, // Issue with bytes32 handling
-      allocation.payoutId,
-      allocation.recurring,
-      allocation.period,
-      allocation.balance,
-      allocation.tokenAddress
+      payoutId,
+      recurring,
+      period,
+      balance,
+      tokenAddress
     )
-    this.closePanel()
+    closePanel()
   }
 
-  onExecutePayout = (accountId, payoutId) => {
-    this.props.api.runPayout(accountId, payoutId)
+  const onExecutePayout = (accountId, payoutId) => {
+    api.runPayout(accountId, payoutId)
```

Comment:
Same thing here. The intermediate function doesn't really get us anything. Let's pass `api.runPayout` instead of `onExecutePayout`.